### PR TITLE
Update Docker Swarm

### PR DESCRIPTION
**Pull request by ChrisChinchilla**

### CODE REVIEW

File: pages/docs/scale/docker_swarm.html [hunk 1]
```diff
@@ -0,0 +1,171 @@
+title: Docker Swarm
+author: Chris Ward
+
```
Comment (chaudum) [inline pages/docs/scale/docker_swarm.html]: please add `description: ...` (html meta description)

File: pages/docs/scale/docker_swarm.html [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+```bash
+gcloud compute firewall-rules create default-ssh --allow tcp:22,icmp --source-range 0.0.0.0/0
+gcloud compute firewall-rules create default-swarm --allow tcp:2376,tcp:3376 --source-range 0.0.0.0/0
+gcloud compute firewall-rules create default-crate --allow tcp:4200,tcp:4300 --source-range 0.0.0.0/0
+```
+
+### Create Master
+
+Create a bash script called _startup-master.sh_ and add the following to it:
```
Comment (chaudum) [inline pages/docs/scale/docker_swarm.html]: I think the markdown parser does not interpret `_` (= code friendly parsing)
you'd need to use single `*`

Comment (ChrisChinchilla) [inline pages/docs/scale/docker_swarm.html]: @chaudum Yup, that keeps throwing me, my editor defaults to '_' (As do most editors…)

File: pages/docs/scale/docker_swarm.html [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+### Creating a Crate Cluster
+
+Now we want to start multiple Crate nodes that form a cluster. Unfortunately, Google Cloud Platform (and other cloud hosting network environments) do not have multicast enabled, so Crate needs to use another method for discovering other Crate nodes.
+
+There are a variety of options including:
+
+- Unicast discovery
+- [DNS discovery](/docs/reference/en/latest/configuration.html#conf-discovery)
+- Using a discovery backend, like [Consul](https://www.consul.io/)
```
Comment (chaudum) [inline pages/docs/scale/docker_swarm.html]: Consul is a backend for Swarm, not at discovery mechanism of Crate

File: pages/docs/scale/docker_swarm.html [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+```
+
+### Creating a Crate Cluster
+
+Now we want to start multiple Crate nodes that form a cluster. Unfortunately, Google Cloud Platform (and other cloud hosting network environments) do not have multicast enabled, so Crate needs to use another method for discovering other Crate nodes.
+
+There are a variety of options including:
+
+- Unicast discovery
+- [DNS discovery](/docs/reference/en/latest/configuration.html#conf-discovery)
```
Comment (chaudum) [inline pages/docs/scale/docker_swarm.html]: DNS discovery is also using Unicast, is is just a mechanism how to the the list of unicast hosts
https://crate.io/docs/reference/en/latest/architecture/shared_nothing.html#discovery

File: pages/docs/scale/docker_swarm.html [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+- Using a discovery backend, like [Consul](https://www.consul.io/)
+
+The simplest is unicast discovery, but it's not the most dynamic. As this documentation is more about Swarm, we will use Unicast discovery.
```
Comment (chaudum) [inline pages/docs/scale/docker_swarm.html]: Unicast can also be dynamic, see DNS or other unicast host providers

Comment (chaudum) [inline pages/docs/scale/docker_swarm.html]: I think the point that you want to make is that you use a static list of unicast hosts (provided via the command line)

File: pages/docs/scale/docker_swarm.html [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+Because Crate is running inside a container, the Crate node will need to publish the IP or hostname of its host. Usually you know on which host you're running your Crate container, so can pass the hostname in the `docker run` command. When using Swarm you don't know where your container will be deployed when you execute the `docker run` command.
+
+Here's our recommended solution:
+
+- Get a list of all remote Docker hosts
+- Assemble a list of unicast hosts from them
+- Calculate the `minimum_master_nodes` setting from the amount of available Docker host (`nodes/2+1`)
```
Comment (chaudum) [inline pages/docs/scale/docker_swarm.html]: maybe you want to reference the documentation of the setting?

File: pages/docs/scale/docker_swarm.html [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+- Iterate over all remote Docker hosts and execute `docker run` explicitly on each host
+
+Create a bash script called _cluster-start.sh_ and add the following:
```
Comment (chaudum) [inline pages/docs/scale/docker_swarm.html]: same here: s/_/*/

File: pages/docs/scale/docker_swarm.html [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+There are a variety of options including:
+
+- Unicast discovery
+- [DNS discovery](/docs/reference/en/latest/configuration.html#conf-discovery)
+- Using a discovery backend, like [Consul](https://www.consul.io/)
+
+The simplest is unicast discovery, but it's not the most dynamic. As this documentation is more about Swarm, we will use Unicast discovery.
+
+Because Crate is running inside a container, the Crate node will need to publish the IP or hostname of its host. Usually you know on which host you're running your Crate container, so can pass the hostname in the `docker run` command. When using Swarm you don't know where your container will be deployed when you execute the `docker run` command.
```
Comment (chaudum) [inline pages/docs/scale/docker_swarm.html]: Not any more since Docker 1.9
https://docs.docker.com/engine/userguide/networking/dockernetworks/

Comment (ChrisChinchilla) [inline pages/docs/scale/docker_swarm.html]: @chaudum I was going to update this and then realised that we still need the bash script steps to get the list of hosts, so I'll just remove the line.

**Comment by ChrisChinchilla** [comment]
@chaudum Do you mind taking another quick look?

**Pull request closed without merging**